### DISCUSSION TITLE
Selection of the finalists

### DISCUSSION THREAD

**Discussion by omichel** [cyberbotics/wrestling#106] [Announcements]
Today at [15:00:00 CEST](https://dateful.com/time-zone-converter?t=15&d=2023-09-19&tz2=Central-European-Summer-Time-CEST), we will select the best 32 teams from the leader board who will participate to the finals.

I will post here the ranking of these 32 selected teams which will be used to organize the finals.

Next week on September 26th between [15:00:00 CEST](https://dateful.com/time-zone-converter?t=15&d=2023-09-26&tz2=Central-European-Summer-Time-CEST) and [18:00:00 CEST](https://dateful.com/time-zone-converter?t=18&d=2023-09-26&tz2=Central-European-Summer-Time-CEST), we will freeze the code of the competitors and use this code to run the finals. Please do not commit any change in your code during this time frame as there is no guarantee it will be taken into account.

**Comment by omichel** [comment]
Here are the 32 finalists:
![image](https://github.com/cyberbotics/wrestling/assets/1264964/bac90c42-1af8-41a7-b116-a7fd1b89aab3)
![image](https://github.com/cyberbotics/wrestling/assets/1264964/84593587-6a25-4eca-815f-38c75a99c5e6)
![image](https://github.com/cyberbotics/wrestling/assets/1264964/768a4b18-c361-43d8-8e06-e40c617879c1)

**Reply by kibergrad** [reply]
@omichel  It appears that Team Humanoid is currently in the first position, and HeavyRotation is not in that position after the queue has ended and is no longer accepting any further triggers. Is that correct?

I recall that you waited during the last ICRA event for the queue to conclude.

**Comment by omichel** [comment]
No, the deadline was 15:00:00, regardless of the queue. And the queue is not blocked at the moment, so that the ranking may still evolve. However, the ranking retained to organize the finals is the one of today 15:00:00 as displayed above. This doesn't mean that HeavyRotation has won the finals. It simply means that HeavyRotation will play its first finals game versus ADWIK while Team Humanoid will play its first final game versus T, etc.

**Reply by kibergrad** [reply]
Thanks for the clarification.

**Comment by omichel** [comment]
Reminder: the code freeze is going to happen in one hour from now. Please do not push any change to your repository between [15:00:00 CEST](https://dateful.com/time-zone-converter?t=15&d=2023-09-26&tz2=Central-European-Summer-Time-CEST) and [18:00:00 CEST](https://dateful.com/time-zone-converter?t=18&d=2023-09-26&tz2=Central-European-Summer-Time-CEST).

**Comment by omichel** [comment]
Due to some problem caused by an update today of the NVIDIA CUDA package on the server machine, we are extending the code freeze deadline by one hour, e.g., until [16:00:00 CEST](https://dateful.com/time-zone-converter?t=16&d=2023-09-26&tz2=Central-European-Summer-Time-CEST).

**Reply by omichel** [reply]
All the controllers have been downloaded. We are going to run the final games next week.

**Comment by omichel** [comment]
The finals table is available [here](https://github.com/cyberbotics/wrestling/blob/main/result-iros2023.md). The first round will be run on October 2nd, 2023. Stay tuned!

**Reply by omichel** [reply]
I did a mistake for the 1/8, 1/4 and 1/2 final games which is now fixed [here](https://github.com/cyberbotics/wrestling/blob/main/result-iros2023.md).

**Reply by kibergrad** [reply]
Hello @omichel,
I hope you're doing well. I wanted to bring up a potential discrepancy in the [results-iros2023.md](https://github.com/cyberbotics/wrestling/blob/main/result-iros2023.md) page regarding the format of the upcoming finals. It seems that there might be conflicting information. The [result.md](https://github.com/cyberbotics/wrestling/blob/main/result.md) and [readme.md](https://github.com/cyberbotics/wrestling#finals-games) mentions that the finals will be in a best-of-five format. However the  [results-iros2023.md](https://github.com/cyberbotics/wrestling/blob/main/result-iros2023.md) states it will be in a best-of-three format. If it's convenient, could you please clarify which format will be used for the finals?

I understand changing the format now might not be possible as the  1/16 finals are approaching and a set of five will  significantly increase the duration of finals. I appreciate your assistance in resolving this matter, and I hope it doesn't bring too much trouble to you. 

Thank you for your understanding and efforts.

**Reply by omichel** [reply]
Oh yes, you are right. We should run the best-of-five format. I will fix this in [result-iros2023.md](https://github.com/cyberbotics/wrestling/blob/main/result-iros2023.md).

**Reply by omichel** [reply]
I just fixed it. Let me know if you spot any other issue with it.

**Reply by kibergrad** [reply]
Sure, I will.
Thanks for your time.

**Comment by AndrejOrsula** [comment]
Thank you very much for the effort required to organize this competition yet again!

Good luck to all the participants in the finals! :smiley:

**Comment by omichel** [comment]
The first round is now complete, check the results at https://github.com/cyberbotics/wrestling/blob/main/result-iros2023.md
Please let me know if you spot any major problem as I didn't visually check everything.

**Reply by adwik-wiz** [reply]
Hi @omichel 

I've watched "HeavyRotation" vs "ADWIK" and noticed that the controllers froze during the 1st and 3rd games of that match  . You mentioned in the [discussion](https://github.com/cyberbotics/wrestling/discussions/112#discussioncomment-7108975) that you would stop the simulation and restart it in such cases . Could you please investigate this issue as discussed?
Thank you.

**Reply by omichel** [reply]
Yes, HeavyRotation is stopping to move after a while, but I believe it is not crashed and this is part of its strategy, so I think it is OK. Moreover, it won, so there is really no reason for re-running these games.

**Reply by adwik-wiz** [reply]
@omichel  My controller "ADWIK" crashed in the first game , which wasn't expected , it shouldn't have stopped

**Reply by omichel** [reply]
I don't see you controller crashing in the first game. It keeps moving all the time. At which time do you believe it crashed?

**Reply by adwik-wiz** [reply]
@omichel at 1:16 after standing up , it should have kept moving , but it didnt move just the hands were moving

**Reply by MJavadZallaghi** [reply]
I noticed that my robot “MechaCheetahs” is frozen in [one of the games](https://webots.cloud/run?version=R2023b&url=https://github.com/cyberbotics/wrestling/blob/IROS2023/worlds/wrestling.wbt&type=competition&context=view&id=IROS2023-A8-3-L) at the second 25, While We have won the game, but “robot being frozen” is not part of our robot control strategy!
I just wanted to mention this issue for your reference in the future.
Thanks in advance.

**Reply by omichel** [reply]
@adwik-wiz: Yes, I noticed that, but the problem here is clearly in your controller, not in our infrastructure as your controller is live and didn't even crashed.

**Reply by omichel** [reply]
@MJavadZallaghi: In your case, it could be two things: either your controller crashed for some reason (which could be a bug in your code, or some external reason) or it didn't crashed but stopped sending commands to the robot for some reason (strategy or bug in your program). Since you won the game, I didn't inspected the reason for this apparent crash. I will investigate only if a controller loose a game while there is a suspicion of crash for it. Only in the case where I identify it crashed for some external reason, I will re-run the game.

**Comment by kibergrad** [comment]
> @omichel at 1:16 after standing up , it should have kept moving , but it didnt move just the hands were moving

Something similar has happened during the past with my controller. And most of the time it happens just after standing up. I don't know the reason but yes, this happens quite sometimes @omichel

**Reply by adwik-wiz** [reply]
@omichel 
Will the matches where the controllers froze be re-run, considering other participants are experiencing the same issue?

**Reply by omichel** [reply]
I don't believe that in this case there is a frozen controller problem, because the robot is clearly not crashed and sending regularly motor commands to the arms of the robot. It is rather a bug in the controller code that in our infrastructure.

**Reply by omichel** [reply]
So, clearly we won't re-run that games.

**Reply by adwik-wiz** [reply]
It's ok if you feel so, but its quite strange on my part as i never saw the bot previously acting like this on webots. 
Thanks for considering my issue.

**Comment by omichel** [comment]
The second round (1/8 finals) is now complete, check the results at https://github.com/cyberbotics/wrestling/blob/main/result-iros2023.md
Please let me know if you spot any major problem as I didn't visually check everything.

**Reply by MJavadZallaghi** [reply]
I checked some matches, and every thing was OK.

**Comment by omichel** [comment]
The 1/4 finals is now complete, check the results at https://github.com/cyberbotics/wrestling/blob/main/result-iros2023.md.

**Comment by omichel** [comment]
The semi-finals are now complete! Check it at https://github.com/cyberbotics/wrestling/blob/main/result-iros2023.md.

**Comment by danielebg** [comment]
Hi @omichel in the 1/4 finals the game 3 of uditech vs HexHowells, From minute 1:38 the HexHowells robot leaves the platform and at one point its counting time does not advance. Was there any error in that simulation?

Thank you

**Reply by omichel** [reply]
This is not a bug, but a limitation of the KO algorithm. It seems that when the two robots were down, out of the platform, HexHowells managed to get higher than uditech, even though it was out of the platform thus stopping its KO countdown temporarily, while uditech kept falling having its countdown running more often than HexHowells. I agree this may seem a bit strange, but if one of the robots had clearly recovered and reached a stable position on the platform, it would certainly have won that game. Obviously, this was not the case and I admit that the determination of the winner is a bit fuzzy in such a case.

**Comment by omichel** [comment]
And the winner is :tada: MechaCheetahs!
Silver medal goes to Toddler and Bronze medal goes to HexHowells.
Congratulation to the winners and to all the participants who really did a great job.
I would like to address a special mention to ijnek who was the only participant using a programming language other than Python, e.g., ROS2.
All the games are available on https://github.com/cyberbotics/wrestling/blob/main/result-iros2023.md.

**Reply by MJavadZallaghi** [reply]
OMG! 
I'm really happy that our team “MechaCheetahs” won this amazing competition. 
All the teams performed well.
Especial thanks to you, dear @omichel for supporting us in this awesome competition.

**Comment by omichel** [comment]
I just reset the main webservice so that you can continue, if you want, to participate to this online competition, although we have no plan yet for another round of finals with a prize.
